.github/workflows/ci.yml

```diff
@@ -20,12 +20,13 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - name: "Checkout the source code"
-        uses: actions/checkout@v3
+        uses: actions/checkout@v4
 
       - name: "Install Python"
-        uses: actions/setup-python@v4
+        uses: actions/setup-python@v5
         with:
           python-version: "3.11"
+          cache: "pip"
```

Comment:
I'm not sure if caching can be used without `requirements.txt`, maybe it's not worth anyway?

Reply:
it works with setup.py too but I forgot to add the params. Thanks.

Reply:
Eventually we plan to move to pyproject.toml now that setup.py is going to be deprecated soon.

Reply:
saves ~1m per job for total of 11 mins across a run. I'll take what I can get.